compiler/ballerina-lang/src/main/resources/compiler.properties

```
@@ -20,6 +20,9 @@
 # Compiler warning messages
 # -------------------------
 
+error.usage.of.async.call.within.lock.is.prohibited=\
+  usage of async call ''{0}'' inside of lock statment is prohibited
```

Can we differentiate between this error being cause due to start vs named worker?

Shall we also reword this to something like
```suggestion
  cannot use an async call inside a lock statement
```
and 
```suggestion
  cannot use a named worker inside a lock statement
```

@hasithaa, @pubudu91, thoughts?

There's a typo in "statment" btw.

---

In case of start the same thing should happen as well right?

---

+1 to reword. I think having two error messages is good.

---

will fix